web/components/pages/pages-list/list-item.tsx

```diff
@@ -0,0 +1,278 @@
+import { observer } from "mobx-react-lite";
+
+export interface IPagesListItem {
+  workspaceSlug: string;
+  projectId: string;
+  page: IPage;
+}
+
+export const PagesListItem: FC<IPagesListItem> = observer((props) => {
+  const { workspaceSlug, projectId, page } = props;
+  // states
+  const [createUpdatePageModal, setCreateUpdatePageModal] = useState(false);
+  const [archivePageModal, setArchivePageModal] = useState(false);
+  // store
+  const {
+    page: { removeFromFavorites, addToFavorites, makePublic, makePrivate, archivePage, restorePage },
+    user: { currentUser },
+    projectMember: { projectMembers },
+  } = useMobxStore();
+  // hooks
+  const { setToastAlert } = useToast();
+
+  const handleCopyText = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    const originURL = typeof window !== "undefined" && window.location.origin ? window.location.origin : "";
+    copyTextToClipboard(`${originURL}/${workspaceSlug}/projects/${projectId}/pages/${page.id}`).then(() => {
+      setToastAlert({
+        type: "success",
+        title: "Link Copied!",
+        message: "Page link copied to clipboard.",
+      });
+    });
+  };
+
+  const handleAddToFavorites = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    addToFavorites(workspaceSlug, projectId, page.id)
+      .then(() => {
+        setToastAlert({
+          type: "success",
+          title: "Success!",
+          message: "Successfully added the page to favorites.",
+        });
+      })
+      .catch(() => {
+        setToastAlert({
+          type: "error",
+          title: "Error!",
+          message: "Couldn't add the page to favorites. Please try again.",
+        });
+      });
+  };
+
+  const handleRemoveFromFavorites = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    removeFromFavorites(workspaceSlug, projectId, page.id)
+      .then(() => {
+        setToastAlert({
+          type: "success",
+          title: "Success!",
+          message: "Successfully removed the page from favorites.",
+        });
+      })
+      .catch(() => {
+        setToastAlert({
+          type: "error",
+          title: "Error!",
+          message: "Couldn't remove the page from favorites. Please try again.",
+        });
+      });
+  };
+
+  const handleMakePublic = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    makePublic(workspaceSlug, projectId, page.id);
+  };
+
+  const handleMakePrivate = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    makePrivate(workspaceSlug, projectId, page.id);
+  };
+
+  const handleArchivePage = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    archivePage(workspaceSlug, projectId, page.id);
+  };
+
+  const handleRestorePage = (e: any) => {
+    e.preventDefault();
+    e.stopPropagation();
+    restorePage(workspaceSlug, projectId, page.id);
+  };
+
+  const handleEditPageModal = (e: any) => {
+    e.preventDefault();
```

Comment:
This doesn't seem to do any logic for editing page, could you check why this is required?

Reply:
This code is work in progress.